### DISCUSSION TITLE
Can't remove stash

### DISCUSSION THREAD

**Discussion by azzamsa** [NeogitOrg/neogit#1169] [Q&A] [answered]
Hi, :wave:

I can't remove my stash, how to debug this error?

https://github.com/NeogitOrg/neogit/assets/17734314/7a82441c-9f75-464b-8c5b-72b74f250707

```
        "neogitorg/neogit",
        event = "VeryLazy",
        commit = "38dd297a905ec6869f4c20ea9184a3e514316e3b",
```

```
🐔 ❯ neovide --version
neovide 0.12.2

🐔 ❯ nvim --version
NVIM v0.9.5
```

```
- **OS Name:**                                     Fedora Linux 39 (Workstation Edition)
- **OS Build:**                                    (null)
- **OS Type:**                                     64-bit
- **GNOME Version:**                               45.3
- **Windowing System:**                            Wayland
- **Kernel Version:**                              Linux 6.6.13-200.fc39.x86_64
```

**Comment by CKolkey** [comment, marked as answer]
Oh, you're hitting "x" to discard, yeah? It's actually (slightly) a bug that this works how it does. It's just trying the discard something from the index here. To discard a stash, use the stash popup to drop it, and I'll see about adding a proper discard here.

**Reply by azzamsa** [reply]
>  To discard a stash, use the stash popup to drop it, 

work like a charm!

**Reply by azzamsa** [reply]
But, in Magit. I can use "x" to discard stash. The same behaviour to discard hunks.

![image](https://github.com/NeogitOrg/neogit/assets/17734314/c1ad1c94-d009-483f-b8fb-58bd2a5fa944)

**Comment by ktsujister** [comment]
> Oh, you're hitting "x" to discard, yeah? It's actually (slightly) a bug that this works how it does. It's just trying the discard something from the index here. To discard a stash, use the stash popup to drop it, and I'll see about adding a proper discard here.

Currently, when you drop stash via stash popup, it drops it immediately without confirmation. Is there any way I can configure it to ask for confirmation before dropping stash via stash popup?

**Reply by CKolkey** [reply]
Yeah, it would make sense to confirm the action if you were not prompted to select something

**Reply by ktsujister** [reply]
It's now fixed with 387df88. Thank you!

**Reply by CKolkey** [reply]
You're welcome :) Thanks for the suggestion
